lib/AST/ASTPrinter.cpp

```diff
@@ -4151,7 +4154,12 @@ void PrintAST::visitLoadExpr(LoadExpr *expr) {
 }
 
 void PrintAST::visitTypeExpr(TypeExpr *expr) {
-  printType(expr->getType());
+  if (auto metaType = expr->getType()->castTo<AnyMetatypeType>()) {
+    // Don't print `.Type` for an expr.
+    printType(metaType->getInstanceType());
+  } else {
+    printType(expr->getType());
+  }
```

Comment:
Just noticed this condition won't actually fail as `castTo` will never return a null type. Additionally it will crash if the `TypeExpr`'s type has not been set yet (which I don't *think* can be the case for the Codable synthesis in particular, but in general we should handle it).

We ought to print the instance type if it has been set, or else fallback to printing the `TypeRepr` instead, e.g:

```cxx
  if (auto ty = expr->getInstanceType()) {
    printType(ty);
  } else {
    expr->getTypeRepr()->print(Printer, Options);
  }
```

I'm happy for this to be done in a follow-up though.